plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -571,10 +571,21 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
                 ImmutableList.of());
     }
 
+    @Override
+    public SchemaTableName getTableName(ConnectorSession session, ConnectorTableHandle table)
+    {
+        if (table instanceof CorruptedIcebergTableHandle corruptedTableHandle) {
+            return corruptedTableHandle.schemaTableName();
+        }
+        return ((IcebergTableHandle) table).getSchemaTableName();
+    }
+
     @Override
     public ConnectorTableMetadata getTableMetadata(ConnectorSession session, ConnectorTableHandle table)
     {
         IcebergTableHandle tableHandle = checkValidTableHandle(table);
+        // This method does not calculate column metadata for the projected columns
+        checkArgument(tableHandle.getProjectedColumns().isEmpty(), "Unexpected projected columns");
```

Comment:
`BaseConnectorTest#testMaterializedViewGracePeriod` for iceberg fails

```
assertThat(query(defaultSession, "TABLE " + viewName)).hasPlan(readFromBaseTables).matches(initialResults);
```

Stack trace:

```
java.lang.IllegalArgumentException: Unexpected projected columns

	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:145)
	at io.trino.plugin.iceberg.IcebergMetadata.getTableMetadata(IcebergMetadata.java:588)
	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorMetadata.getTableMetadata(ClassLoaderSafeConnectorMetadata.java:272)
	at io.trino.tracing.TracingConnectorMetadata.getTableMetadata(TracingConnectorMetadata.java:252)
	at io.trino.metadata.MetadataManager.getTableMetadata(MetadataManager.java:463)
	at io.trino.tracing.TracingMetadata.getTableMetadata(TracingMetadata.java:287)
	at io.trino.sql.planner.assertions.TableScanMatcher.detailMatches(TableScanMatcher.java:61)
	at io.trino.sql.planner.assertions.PlanMatchPattern.detailMatches(PlanMatchPattern.java:921)
	at io.trino.sql.planner.assertions.PlanMatchingVisitor.matchLeaf(PlanMatchingVisitor.java:148)
	at io.trino.sql.planner.assertions.PlanMatchingVisitor.visitPlan(PlanMatchingVisitor.java:109)
	at io.trino.sql.planner.assertions.PlanMatchingVisitor.visitPlan(PlanMatchingVisitor.java:35)
	at io.trino.sql.planner.plan.PlanVisitor.visitTableScan(PlanVisitor.java:74)
```